File: examples/python/l1-resources/project.py
from aws_cdk import Stack
from constructs import Construct

from awscdk_resources_mongodbatlas import CfnProject

class CdkTestAppPython2Stack(Stack):

  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id, **kwargs)

    project_name_var = self.node.try_get_context('project_name')
    org_id_var = self.node.try_get_context('org_id')
    profile_name_var = self.node.try_get_context('profile')

    project_l1 = CfnProject(self, "Project-py-l1",
                            name=project_name_var,
                            org_id=org_id_var,
                            profile=profile_name_var)

File: examples/java/l1-resources/Project.java
package com.myorg;

import java.util.List;
import org.jetbrains.annotations.NotNull;
import software.constructs.Construct;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import org.mongodb.awscdk.resources.mongodbatlas.*;
import software.constructs.Node;


public class CdkJavaStack extends Stack {
    final static String ORG_ID = "ORG_ID";
    final static String PROFILE = "PROFILE";
    final static String PROJECT_NAME = "PROJECT_NAME";


    public CdkJavaStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        final String profile = getContextProp(this.getNode(), PROFILE);
        final String projectName = getContextProp(this.getNode(), PROJECT_NAME);
        final String orgId = getContextProp(this.getNode(), ORG_ID);

        final CfnProject myProj = CfnProject.Builder.create(this, "myProject")
        .name(projectName)
        .orgId(orgId)
        .profile(profile)
        .build();
    }

    public String getContextProp(final Node pNode, final String pPropName){
        @NotNull final Object obj = pNode.tryGetContext(pPropName);
        if (obj.toString().isEmpty()) {
            return "";
        }
        return obj.toString();
    }
}
